packages/core/src/external/carequality/dq/process-inbound-dq.ts

```diff
@@ -1,12 +1,49 @@
+import { DocumentReference } from "@medplum/fhirtypes";
 import { InboundDocumentQueryReq, InboundDocumentQueryResp } from "@metriport/ihe-gateway-sdk";
-import { constructDQErrorResponse, IHEGatewayError, XDSRegistryError } from "../error";
-import { findDocumentReferences } from "./find-document-reference";
+import axios from "axios";
+import {
+  createUploadFilePath,
+  createUploadMetadataFilePath,
+} from "../../../domain/document/upload";
+import { Config } from "../../../util/config";
+import { out } from "../../../util/log";
+import { capture } from "../../../util/notifications";
+import { sizeInBytes } from "../../../util/string";
+import { uuidv7 } from "../../../util/uuid-v7";
+import { createAndUploadMetadataFile } from "../../aws/lambda-logic/document-uploader";
+import { S3Utils } from "../../aws/s3";
+import {
+  IHEGatewayError,
+  XDSRegistryError,
+  XDSUnknownPatientId,
+  constructDQErrorResponse,
+} from "../error";
+import { decodePatientId, findDocumentReferences } from "./find-document-reference";
+
+const region = Config.getAWSRegion();
+const s3Utils = new S3Utils(region);
+const api = axios.create();
+const bucket = Config.getMedicalDocumentsBucketName();
 
 export async function processInboundDocumentQuery(
-  payload: InboundDocumentQueryReq
+  payload: InboundDocumentQueryReq,
+  apiUrl: string
 ): Promise<InboundDocumentQueryResp> {
   try {
-    const documentContents = await findDocumentReferences(payload);
+    const id_pair = decodePatientId(payload.externalGatewayPatient.id);
+
+    if (!id_pair) {
+      throw new XDSUnknownPatientId("Patient ID is not valid");
+    }
+    const { cxId, id: patientId } = id_pair;
+
+    let documentContents;
+    documentContents = await findDocumentReferences(payload);
```

Comment:
super nit: can assign the variable along the declaration

not so nit: can avoid variables (constants are our friends) by creating a function that returns the doc contents and has the logic to create the CCD if nothing else found